src/sentry/rules/processor.py

```diff
@@ -111,30 +109,18 @@ def apply_rule(self, rule):
             return
 
         now = timezone.now()
-        if passed and rule_status.status == GroupRuleStatus.INACTIVE:
-            # we only fire if we're able to say that the state has changed
-            GroupRuleStatus.objects.filter(
-                id=rule_status.id,
-                status=GroupRuleStatus.INACTIVE,
-            ).update(
-                status=GroupRuleStatus.ACTIVE,
-                last_active=now,
-            )
-            rule_status.last_active = now
-            rule_status.status = GroupRuleStatus.ACTIVE
-        elif not passed and rule_status.status == GroupRuleStatus.ACTIVE:
-            # update the state to suggest this rule can fire again
-            GroupRuleStatus.objects.filter(
-                id=rule_status.id,
-                status=GroupRuleStatus.ACTIVE,
-            ).update(status=GroupRuleStatus.INACTIVE)
-            rule_status.status = GroupRuleStatus.INACTIVE
-        elif passed:
-            GroupRuleStatus.objects.filter(
-                id=rule_status.id,
-                status=GroupRuleStatus.ACTIVE,
+        freq_offset = now - timedelta(minutes=frequency)
+
+        if (passed and status.last_active and
+                status.last_active > freq_offset):
```

Comment:
Would it make sense to check the last active timestamp first to avoid extra work (e.g. hitting TSDB to check condition criteria, etc) and short circuit `return false` if the frequency threshold isn't met?